server/routes/creatingReports/checkYourAnswers.ts

```diff
@@ -11,28 +12,46 @@ export default class CheckAnswerRoutes {
     private readonly draftReportService: DraftReportService,
     private readonly offenderService: OffenderService,
     private readonly systemToken: SystemToken,
-    private readonly locationService: LocationService
+    private readonly locationService: LocationService,
+    private readonly nomisMappingService: NomisMappingService
   ) {}
 
   public view = async (req: Request, res: Response): Promise<void> => {
+    const token = await this.systemToken(res.locals.user.username)
     const { bookingId } = req.params
     const {
+      id,
       form = {},
       incidentDate,
       agencyId: prisonId,
     } = await this.draftReportService.getCurrentDraft(req.user.username, parseInt(bookingId, 10))
+
+    // At this point the reportStatus may be 'incomplete' (because of the validation rules) if only the nomis locationId is present.
+    // This scenario would occur where a user is already at /check-your-answers when the new code to use incidentLocationId is deployed
+    // To pass validation we need the equivalent dpsLocationId persisted to the db as incidentLocationId.
+    // The following 'if-block' does that.
+
```

Comment:
if at the /check-your-answers page only the prisonApi id exists, look up the equivalent locationsApi id and save it to db, This will ensure validation does not fail